packages/devtools_flutter/lib/src/connect_screen.dart

```diff
@@ -98,9 +103,26 @@ class _ConnectScreenBodyState extends State<ConnectScreenBody> {
     );
   }
 
-  void connect() {
-    // TODO(https://github.com/flutter/devtools/issues/1112): Connect to a VM.
-    Navigator.pushNamed(context, '/connected');
+  Future<void> connect([_]) async {
+    final connected = await service.FrameworkCore.initVmService(
+          '',
```

Comment:
is there a way to do an exclusive or with method params? It seems like exactly one of `url` and `explicitUri` are needed for the `initVmService` method. If not, maybe we should change the parameters of `initVmService` so that both `url` and `explicitUri` are optional named. It seems odd that the required parameter can be an empty string.

Reply:
It is possible to do this:

```
void method({param1, param2}) {
  assert((param1 == null) != (param2 == null));
}
```

A change like that is outside the scope of this PR, though.